src/tools/bin/nextstep_from_period.sh

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env bash
+#
+# Copyright 2017-2018 Axel Huebl, Ilja Goethel
+#
+# This file is part of PIConGPU.
+#
+# PIConGPU is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# PIConGPU is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with PIConGPU.
+# If not, see <http://www.gnu.org/licenses/>.
+#
+
+set -euf -o pipefail
+
+helpText()
+{
+    echo "
+    Usage: $0 period lastStep [currentStep]
+
+    Script returns next timestep specified by the period syntax
+    Takes three positional arguments:
+     - a string representing period-syntax, like 3000,12000:15000:1000,1337:1337
+     - and the final timestep (convertible to int)
+     - the current timestep (convertible to int)
+
+    and prints the next timestep to stdout"
+}
+
+# now we parse cmd params: first look for -h, then case depending on $#
+
+for p in $@; do
+    if [ "$p" = "-h" ] || [ "$p" = "--help" ]; then
+        helpText
+        exit 0
+    fi
+done
+
+case $# in
+    0 | 1)
+        printf "\n *** Not enough arguments! *** \n"
+        helpText
+        exit 1
+    ;;
+    2)
+        period="$1"
+        lastStep="$2"
+        currentStep=0
+    ;;
+    3)
+        period="$1"
+        lastStep="$2"
+        currentStep="$3"
+    ;;
+    *)
+        printf "\n *** Too many arguments! *** \n"
+        helpText
+        exit 1
+    ;;
+esac
+
+splitSections()
+{
+    # split a string on occurences of ','
+    IFS=','
+    for s in $1; do
+        echo $s
+    done
+}
+expandPeriods()
+{
+    # split a string on occurences of ':' and always return a 3-tuple
```

Comment:
It does not return a 3-tuple - it internally splits the given string into a 3-typle (with default values if needed) and returns a sequence of checkpoint time steps. 
Please specify the comment.

Reply:
will fix